tests/src/test/scala/common/WskTestHelpers.scala

```diff
@@ -31,6 +31,55 @@ import spray.json._
 import java.time.Instant
 
 /**
+ * An arbitrary response of a whisk action. Includes the result as a JsObject as the
+ * structure of "result" is not defined.
+ */
+case class ActivationResponse(result: Option[JsObject], status: String, success: Boolean)
+
+object ActivationResponse extends DefaultJsonProtocol {
+    implicit val serdes = jsonFormat3(ActivationResponse.apply)
+}
+
+/**
+ * Activation record as it is returned by the CLI.
+ */
+ case class ActivationResult (
+        activationId: String,
+        logs: Option[List[String]],
+        response: ActivationResponse,
+        start: Instant,
+        end: Option[Instant], // This field is optional in responses of REST calls.
+        duration: Option[Long], // This field is optional in responses of REST calls.
```

Comment:
Could we not make those optional but have them deserialize to default values? I think if `end` is ommitted, the Rest API would give it an `Instant.EPOCH` while the CLI omits it. Duration I think is always `0` if ommitted.